test/gtest/uct/test_uct_perf.cc

```diff
@@ -244,4 +249,40 @@ UCS_TEST_P(test_uct_loopback, envelope)
     test_execute(UCX_PERF_TEST_FLAG_LOOPBACK);
 }
 
-UCT_INSTANTIATE_TEST_CASE(test_uct_loopback);
+UCT_INSTANTIATE_NO_CUDA_TEST_CASE(test_uct_loopback);
+
+class test_uct_loopback_cuda : public test_uct_perf {
+public:
+    const std::vector<std::vector<ucs_memory_type_t>>& mem_type_pairs() {
+        static std::vector<std::vector<ucs_memory_type_t>> result;
+
+        if (result.empty()) {
+            std::vector<ucs_memory_type_t> input;
+
+            input.push_back(UCS_MEMORY_TYPE_HOST);
+            input.push_back(UCS_MEMORY_TYPE_CUDA);
+
+            result = ucs::make_pairs(input);
+        }
+
+        return result;
+    }
+};
+
+UCS_TEST_P(test_uct_loopback_cuda, envelope)
+{
+    if (has_transport("gdr_copy")) {
+        test_execute(UCX_PERF_TEST_FLAG_LOOPBACK, UCS_MEMORY_TYPE_HOST,
```

Comment:
1. pls add comment - is it because gdr_copy supports only host->cuda?
2. i would move this logic to "mem_type_pairs" to separate pairs generation from test run. we can get rid of caching "result", it's not so important for performance.